test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -101,13 +100,9 @@ public class TestFunctionHost : IDisposable
                 })
                 .UseStartup<TestStartup>();
 
-            // TODO: https://github.com/Azure/azure-functions-host/issues/4876
-            _testServer = new TestServer(builder) { AllowSynchronousIO = true };
+            _testServer = new TestServer(builder) { BaseAddress = new Uri("https://localhost/") };
 
-            HttpClient = new HttpClient(new UpdateContentLengthHandler(_testServer.CreateHandler()))
```

Comment:
This handler was inadvertently reading the request content too early, which was masking this `AllowSynchronousIO` breaking change. Once I removed it, I was able to write a test that failed. To compensate, I've explicitly added Content-Length to the requests that needed it in the tests. We were already doing this in some places.